app/search/views.py

```diff
@@ -8,55 +9,66 @@
 from config.jinja2 import qs_remove_value, qs_toggle_value
 from django.template.response import TemplateResponse
 
-from .buckets import BucketKeys, get_buckets_for_display
+from .buckets import CATALOGUE_BUCKETS, BucketKeys
+from .constants import Sort
 
 
 def catalogue_search_view(request):
     template = "search/catalogue.html"
+    bucket_list = copy.deepcopy(CATALOGUE_BUCKETS)
+    default_group = BucketKeys.TNA.value
+    default_sort = Sort.RELEVANCE.value  # sort includes ordering
+    RESULTS_PER_PAGE = 20  # max records to show per page
+    PAGE_LIMIT = 500  # max page number that can be queried
+
     context: dict = {
         "levels": TNA_LEVELS,
         "closure_statuses": CLOSURE_STATUSES,
         "collections": COLLECTIONS,
     }
-    results_per_page = 20
-    page = int(request.GET.get("page", 1))
-    sort_order = request.GET.get("sort", "").split(":")
-    sort = sort_order[0] if sort_order else ""
-    order = sort_order[1] if len(sort_order) > 1 else ""
 
-    current_bucket_key = request.GET.get("group") or BucketKeys.TNA
-    # filter records for a bucket
-    params = {"filter": f"group:{current_bucket_key}"}
+    try:
+        page = int(request.GET.get("page", 1))
+        if page < 1:
+            raise ValueError
+    except (ValueError, KeyError):
+        # graceful degradation, fallback
+        page = 1
```

Comment:
This is a question for more people, but I would expect that if someone had `page=foobar` in the URL (which they could only really do by modifying it themselves) then I would expect to see an error page and not just the first page of results.

Reply:
I've noticed the page is not handled. Looking into it.